### PR TITLE
Adding custom context processor

### PR DESCRIPTION
**Pull request by saadyousafarbi**
Adding custom context processor

* Custom processor ```context_processor.site_processor``` will return dictionary settings and site object for project.
* These objects will be available to all templates present in project.

### CODE REVIEW

File: autoshop/core/templates/base.html [hunk 1]
```diff
@@ -19,6 +19,13 @@ <h3 class="text-center">A website with information about automobiles for buying/
     {% block content %}{% endblock content %}
     </body>
 
+    <div class="footer">
+        <p>
+            Related Website:
```
Comment (zubair-arbi) [inline autoshop/core/templates/base.html]: See djangos `trans` template tag

File: autoshop/core/templates/base.html [hunk 2]
```diff
@@ -19,6 +19,13 @@ <h3 class="text-center">A website with information about automobiles for buying/
...
+            <b> (PakWheels) </b>
+            <a href='{{PakWheels_site}}' target="_blank">Visit Site</a>
```
Comment (zubair-arbi) [inline autoshop/core/templates/base.html]: need spaces

File: autoshop/analytics/processor.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+def related_website(request):
+
+    related_site = {'PakWheels_site': 'https://www.pakwheels.com/'}
```
Comment (zubair-arbi) [inline autoshop/analytics/processor.py]: this is not what we wanted. Just pass django's settings and site object directly

File: autoshop/analytics/processor.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+def related_website(request):
```
Comment (zubair-arbi) [inline autoshop/analytics/processor.py]: This should be in `autoshop` app or in `core` app as its a site wide context processor.
Also rename it to `site_processor`.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Please fix these suggstions

File: autoshop/core/site_processor.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.conf import settings
+from django.contrib.sites.models import Site
+
+
+def setting_and_site_context(request):
```
Comment (zubair-arbi) [inline autoshop/core/site_processor.py]: rename the file to `context_processors` and rename this method to `site_processor`

File: autoshop/core/templates/base.html [hunk 6]
```diff
@@ -19,6 +19,7 @@ <h3 class="text-center">A website with information about automobiles for buying/
     {% block content %}{% endblock content %}
     </body>
 
+
```
Comment (zubair-arbi) [inline autoshop/core/templates/base.html]: remove this change

File: autoshop/core/site_processor.py (hunk 5 above)
Comment (zubair-arbi) [inline autoshop/core/site_processor.py]: Also add docstring like:
```
"""
Provide site object and common project settings across all project in context.
"""
```

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Please update the PR description and mention what variable will be provided site wide in request context.

**Comment by zubair-arbi** [comment]
Please squash your commits into a single one. Then wait for my PR https://github.com/saadyousafarbi/auto-shop/pull/10 to merge, that PR will change the project structure a bit.
After that update your local master and rebase your branch onto it.

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍

**Comment by saadyousafarbi** [comment]
Last commit 3bfbfa8eaaa4e2c5d9326754a640a8adf40c0943

**Pull request merged**